### PR TITLE
Add ip network exist function to agent_persist_mixin.py

### PR DESCRIPTION
**Pull request by elyousfi5**

### CODE REVIEW

**Review by amine3 — DISMISSED** [review]

File: src/ostorlab/agent/mixins/agent_persist_mixin.py [hunk 1]
```diff
@@ -225,3 +225,39 @@ def add_ip_network(
         else:
             member_value = ip_range.exploded
         return self.set_add(key, member_value)
+
+    def ip_network_exist(
```
Comment (3asm) [inline src/ostorlab/agent/mixins/agent_persist_mixin.py:229]:
```suggestion
    def ip_network_exists(
```

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/agent/mixins/agent_persist_mixin.py [hunk 2]
```diff
@@ -225,3 +225,39 @@ def add_ip_network(
...
+        key: Union[bytes, str],
+        ip_range: Union[ipaddress.IPv6Network, ipaddress.IPv4Network],
+        value: Optional[
+            Callable[
+                [Union[ipaddress.IPv6Network, ipaddress.IPv4Network]], Union[bytes, str]
+            ]
+        ] = None,
+    ) -> bool:
+        """
+        Returns False if a network have never been persisted before, else it's returns True.
```
Comment (PiranhaSa) [inline src/ostorlab/agent/mixins/agent_persist_mixin.py:240]:
```suggestion
        Returns False if a network have never been persisted before, else it returns True.
```

File: src/ostorlab/agent/mixins/agent_persist_mixin.py [hunk 3]
```diff
@@ -225,3 +225,39 @@ def add_ip_network(
         else:
             member_value = ip_range.exploded
         return self.set_add(key, member_value)
+
+    def ip_network_exists(
```
Comment (benyissa) [inline src/ostorlab/agent/mixins/agent_persist_mixin.py:229]: The prefix "is_" suggests that the function returns a boolean value, which can enhance readability and make the purpose of the function more explicit.
```suggestion
    def is_ip_network_exists(
```

File: tests/agent/mixins/agent_persist_mixin_test.py [hunk 4]
```diff
@@ -166,3 +166,34 @@ async def testAgentPersistMixinCheckIpRangeExist_withCallableKey_returnTrue(
...
         is False
     )
+
+
+@pytest.mark.parametrize("clean_redis_data", ["redis://localhost:6379"], indirect=True)
+@pytest.mark.asyncio
+@pytest.mark.docker
+async def testAgentPersistMixinCheckIpNetworkExist_whenIpRangeIsCovered_returnTrue(
+    mocker, redis_service, clean_redis_data
+):
```
Comment (rabsondev) [inline tests/agent/mixins/agent_persist_mixin_test.py:176]: Typing!

File: src/ostorlab/agent/mixins/agent_persist_mixin.py (hunk 3 above)
Comment (benyissa) [inline src/ostorlab/agent/mixins/agent_persist_mixin.py:229]: Consider employing this method within the `add_ip_network` function to eliminate redundancy.

**Review by rabsondev — CHANGES_REQUESTED** [review]

File: src/ostorlab/agent/mixins/agent_persist_mixin.py (hunk 3 above)
Comment (elyousfi5) [reply src/ostorlab/agent/mixins/agent_persist_mixin.py:229]: The add_ip_network role is to add it if its not there, what i want is just to check if it exists. for consistency i prefer doing this separately.

**Review by 3asm — APPROVED** [review]

File: src/ostorlab/agent/mixins/agent_persist_mixin.py (hunk 3 above)
Comment (3asm) [reply src/ostorlab/agent/mixins/agent_persist_mixin.py:229]: Convention is `is_blue`, `is_red` and `file_exists`. Not file `is_file_exists`.

**Pull request merged**